### PR TITLE
[CORE-668,CORE-669] - Add health monitor flags

### PR DESCRIPTION
**Pull request by clemire**
### Changelist
Add 2 flags:
- configure # of seconds before unhealthy daemon causes a panic
- enable panicking behavior or log error when daemon health check fails

### Test Plan
- unit tests for daemonFlags
- tested both flags with localnet.

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

### CODE REVIEW

File: protocol/app/app.go [hunk 1]
```diff
@@ -183,10 +183,6 @@ import (
 var (
 	// DefaultNodeHome default home directories for the application daemon
 	DefaultNodeHome string
-
```
Comment (clemire) [inline protocol/app/app.go:186]: This constant is now defined within daemon flags, not needed here.

**Review by lcwik — APPROVED** [review]

File: protocol/daemons/flags/flags.go [hunk 2]
```diff
@@ -112,6 +121,16 @@ func AddDaemonFlagsToCmd(
 		"Socket address for the price daemon to send updates to, if not set "+
 			"will establish default location to ingest price updates from",
 	)
+	cmd.Flags().Bool(
```
Comment (ttl33) [inline protocol/daemons/flags/flags.go:124]: nit: the method comment says `required flags to instantiate a server and client for price update`. Can we update the comment to be less specific to price update since this is shared now?

File: protocol/daemons/flags/flags.go [hunk 3]
```diff
@@ -74,7 +81,9 @@ func GetDefaultDaemonFlags() DaemonFlags {
 	if defaultDaemonFlags == nil {
 		defaultDaemonFlags = &DaemonFlags{
 			Shared: SharedFlags{
-				SocketAddress: "/tmp/daemons.sock",
+				SocketAddress:                 "/tmp/daemons.sock",
+				DisablePanicOnDaemonFailure:   false,
```
Comment (ttl33) [inline protocol/daemons/flags/flags.go:85]: I think we use `Enabled` (as opposed to disabled) throughout our codebase. For consistency, can we use this format?

```suggestion
				PanicOnDaemonFailureEnabled:   true,
```

File: protocol/daemons/flags/flags.go [hunk 4]
```diff
@@ -9,7 +9,9 @@ import (
 // List of CLI flags for Server and Client.
 const (
 	// Flag names
-	FlagUnixSocketAddress = "unix-socket-address"
+	FlagUnixSocketAddress             = "unix-socket-address"
+	FlagPanicOnDaemonFailureEnabled   = "panic-on-daemon-failure-enabled"
+	FlagMaximumDaemonUnhealthySeconds = "maximum-daemon-unhealthy-seconds"
```
Comment (ttl33) [inline protocol/daemons/flags/flags.go:14]:
```suggestion
	FlagMaxDaemonUnhealthySeconds = "max-daemon-unhealthy-seconds"
```

File: protocol/daemons/flags/flags.go [hunk 5]
```diff
@@ -28,6 +30,10 @@ const (
 type SharedFlags struct {
 	// SocketAddress is the location of the unix socket to communicate with the daemon gRPC service.
 	SocketAddress string
+	// PanicOnDaemonFailureEnabled toggles whether the daemon should panic on failure.
+	PanicOnDaemonFailureEnabled bool
+	// MaximumDaemonUnhealthySeconds is the maximum allowable duration for which a daemon can be unhealthy.
+	MaximumDaemonUnhealthySeconds uint32
```
Comment (ttl33) [inline protocol/daemons/flags/flags.go:36]:
```suggestion
	MaxDaemonUnhealthySeconds uint32
```

File: protocol/app/app.go [hunk 6]
```diff
@@ -613,7 +610,10 @@ func New(
 		if daemonFlags.Liquidation.Enabled {
 			app.LiquidationsClient = liquidationclient.NewClient(logger)
 			go func() {
-				app.MonitorDaemon(app.LiquidationsClient, MaximumDaemonUnhealthyDuration)
+				app.MonitorDaemon(
+					app.LiquidationsClient,
+					time.Duration(daemonFlags.Shared.MaxDaemonUnhealthySeconds)*time.Second,
```
Comment (ttl33) [inline protocol/app/app.go:615]: nit: given that this is common across all daemons, you could calc this once and reuse.

File: protocol/daemons/server/types/health_checker.go [hunk 7]
```diff
@@ -83,26 +83,26 @@ func (u *healthCheckerMutableState) ReportFailure(now time.Time, err error) time
...
-	// pollFrequency is the frequency at which the health checkable service is polled.
+	// pollFrequency is the frequency at which the health-checkable service is polled.
 	pollFrequency time.Duration
 
-	// timer triggers a health check poll for a health checkable service.
+	// timer triggers a health check poll for a health-checkable service.
 	timer *time.Timer
 
-	// maxAcceptableUnhealthyDuration is the maximum acceptable duration for a health checkable service to
+	// maxAcceptableUnhealthyDuration is the maximum acceptable duration for a health-checkable service to
```
Comment (ttl33) [inline protocol/daemons/server/types/health_checker.go:100]: nit: can we use `maxDaemonUnhealthyDuration` to be consistent w the flag name? Feels a bit confusing to go from `maxDaemonUnhealthyDuration` to `maxAcceptableUnhealthyDuration`

File: protocol/daemons/server/types/health_monitor.go [hunk 8]
```diff
@@ -64,13 +64,29 @@ func (ms *healthMonitorMutableState) Stop() {
...
 
-// RegisterHealthChecker registers a new health checker for a health checkable with the health monitor. The health
+// RegisterHealthChecker registers a new health checker for a health-checkable with the health monitor. The health
 // checker is lazily created using the provided function if needed. This method is synchronized. It returns an error if
 // the service was already registered.
 func (ms *healthMonitorMutableState) RegisterHealthChecker(
 	checkable types.HealthCheckable,
 	lazyHealthCheckerCreator func() *healthChecker,
 ) error {
+	stopService := false
```
Comment (ttl33) [inline protocol/daemons/server/types/health_monitor.go:74]: hmm, is this change intended to be part of this PR? thought this PR was strictly adding the flags.

File: protocol/daemons/server/types/health_checker.go (hunk 7 above)
Comment (clemire) [reply protocol/daemons/server/types/health_checker.go:100]: Good point!

File: protocol/daemons/server/types/health_monitor.go (hunk 8 above)
Comment (clemire) [reply protocol/daemons/server/types/health_monitor.go:74]: Ah, looks like I mixed up some changes I meant to add to the other health monitoring PR.

**Review by ttl33 — APPROVED** [review]
